src/image-generation-service.ts

```diff
@@ -94,6 +105,32 @@ export class ImageGenerationService {
     }
     Logger.log('Finished generating.');
   }
+  /**
+   * For a given string & regex return the match groups if they exist else null
+   */
+  getRegexMatchGroups(str, regex) {
+    const regexMatch = str.match(regex);
+    if (regexMatch !== null) {
+      return regexMatch.groups;
+    }
+    return null;
+  }
+  /**
+   * Inject the values from the object into the prompt and return.
+   *
+   * A prompt can have placeholders for keys in the object. For example:
+   * "A photo of a ${city} in sharp, 4k".
+   * If an object with { 'city': 'London' } is provided it will replace the
+   * placeholder and return "A photo of a London in sharp, 4k".
+   */
+  createPrompt(obj) {
+    let prompt = CONFIG['ImgGen Prompt'];
+    for (const [key, value] of Object.entries(obj)) {
+      const regex = new RegExp('(\\${' + key + '})', 'gi');
+      prompt = prompt.replaceAll(regex, value);
```

Comment:
I would consider using a simple match, you don't need regex to replace '${' + name + '}' (`prompt.replaceAll('${' + name + '}', value);`)

Reply:
Another good callout. I've updated this in the latest code change. Thanks